google-cloud-spanner/src/main/java/com/google/cloud/spanner/AbstractResultSet.java

```diff
@@ -1122,117 +1081,6 @@ static NullPointerException throwNotNull(int columnIndex) {
         "Cannot call array getter for column " + columnIndex + " with null elements");
   }
 
-  /**
```

Comment:
@olavloite I removed this optimized Int64 and Float64. Do you think this will impact performance significantly?

In this constructor we were doing `O(n)` compute and using a `bitset` for `nulls`. The difference here will be basically be in memory usage, where here we were using `O(1)` and with the new implementation we copy the array into a list (so `O(n)`). However, I wonder how big is `n` usually and whether we should worry about this.

Reply:
I do worry about the additional memory consumption here, because there are more changes in this PR that affect memory consumption than only the removal of the optimized array decoding of `Int64` and `Float64`. So I do think it is worth checking what effect this has. The other things that affect memory consumption in this PR:
1. Instead of keeping a reference to an `Object` for each column/row combination, we now create a `Value` object. The object itself is not big, but will need to be garbage collected afterwards. For large result sets this could be a significant number of additional objects.
2. Most array types used `Lists.transform` for decoding. That is virtually free as long as the column is not retrieved by the application. If a user executes a select that includes an array column that is not used by the application, it will now always be decoded for each row that is consumed, regardless of whether the column itself is consumed.
3. The optimized `Int64` and `Float64` array implementations are removed.

Also, I think that the current implementations for decoding `Array<Int64>` and `Array<Float64>` to primitive arrays will fail if one of the array elements is null.

I don't worry about the additional CPU cycles for the decoding itself, it's more the additional overhead that is created mainly for the garbage collection of the additional objects that we create.